File: java_projects/glue_code/commons-graph/src/main/java/org/apache/commons/graph/flow/FlowWeightedEdgesBuilder.java
package org.apache.commons.graph.flow;

import org.apache.commons.graph.Mapper;

public interface FlowWeightedEdgesBuilder<V, WE> {
  <W, M extends Mapper<WE, W>> FromHeadBuilder<V, WE, W> whereEdgesHaveWeights(M weightedEdges);
}

File: data/recomposed_projects/llama-3-70b-instruct/body/commons-graph/src/main/org/apache/commons/graph/flow/FlowWeightedEdgesBuilder.py
from __future__ import annotations
import re
from abc import ABC
import io
import typing
from typing import *
from src.main.org.apache.commons.graph.Mapper import *
from src.main.org.apache.commons.graph.flow.FromHeadBuilder import *


class FlowWeightedEdgesBuilder(ABC):

    def whereEdgesHaveWeights(
        self, weightedEdges: typing.Any
    ) -> FromHeadBuilder[typing.Any, typing.Any, typing.Any]:
        return FromHeadBuilder(self, weightedEdges)
